### DISCUSSION TITLE
Etsy Shop is closed but Shop API returns success

### DISCUSSION THREAD

**Discussion by IamSantosh** [etsy/open-api#772] [Q&A]
Hi there,

I am facing one issue regarding API for one of the Shop registered to our Etsy App.

Shop URL : https://www.etsy.com/shop/RedsEmbroideryShop

This shop was registered to our app a few days back and everything worked perfectly unless a few days back, according to the user, he closed his shop. Now on opening the Shop URL, this page appears
![Screenshot from 2022-12-23 10-59-53](https://user-images.githubusercontent.com/19526862/209281353-93574e12-cc29-4152-8610-31e4347fd519.png)

But on hitting the Shop API, it returns success response. 
![oie_zaAfD1rsHjVd](https://user-images.githubusercontent.com/19526862/210038750-4c7b42f0-5e80-42be-bc66-6c51eab9ec36.png)

I also used User and listing API and both of them returned success responses.

Is there any way or API to find out wether the Shop is closed or not?

Please help me out.

**Comment by Mad4F1** [comment]
I have discovered a similar issue, but I personally think that the API *should* return back results for closed shops - it should just mark their status correctly.

To illustrate my point, the issue that I've spotted is that for a temporarily closed shop, the API won't allow you to retrieve all Listings. The Listings in theory *should* be active, but none are returned. This is when using "getListingsByShop". 

However, if you try and access just one of the listings directly (which doesn't require AUTH, so this could be even trying to access someone else's listings!), then the Listing is returned with a state of "active" (even though it doesn't get returned for "getListingsByShop").

There are valid reasons why an authorised request *should* return info about listings, whether the shop is closed or not. For example, someone might temporarily close the shop so that they can do maintenance. However, if they can't access all the listings, without first knowing the ListingId, then this will be a problem.

Also, the enums for state don't seem correct. When I entered an invalid state, I get the following error message:
"Invalid value (all) for enum(0, 1, 2, 3, 4, 5, active, inactive, sold_out, draft, removed, expired, edit)"

However:
1) Enum 4 doesn't work (gives a server error), as does "removed" - so I guess "4" corresponds to "removed". 
2) There are less enum numbers than there are enums (6 enum numbers versus 7 options). So which enum is correct?

Should there be an extra state for when a shop is temporarily closed? Or should the listing still be Active, and the Shop itself is closed?

**Comment by emergingdzns** [comment]
We also are having an issue here. We have some shops in our records we know are closed but the getShop endpoint returns data as if the shop is still open. The only way to tell if they are closed is to call the findActiveListingsByShop endpoint and look for the erorr "No shop found or shop is not available." This is not an efficient way to tell but at the moment, it's all we have.

I'd like to suggest that Etsy change the getShop endpoint to have a status field for `open` or `closed`.

**Comment by Gareth064** [comment]
@IamSantosh @Mad4F1 @emergingdzns - I have reproduced this issue using the shop in the OP's screenshot (RedsEmbroideryShop).

I have emailed Etsy asking for them to look into and to respond to this GitHub discussion with what they find.

For now, since there is no flag on the record to say if it's opened or closed, I wonder if it's possible to assume that if the shop is closed based on some cobbled together logic. For example, this shop has no active listings, or any shop policies at all.

![image](https://user-images.githubusercontent.com/22307223/231215307-97cd187b-0fa1-4cc2-9253-44a49cf72723.png)

**Comment by Gareth064** [comment]
Interestingly, here is another shop with a similar profile in terms of what is returned from the API [Lavender](https://www.etsy.com/shop/lavender) but it sows a different message when you browse to it on Etsy

![image](https://user-images.githubusercontent.com/22307223/231216664-f76ce25b-817f-4c4c-9802-dbd32034203b.png)
